src/lib.rs

```diff
@@ -568,6 +568,19 @@ impl Library {
     }
 
     fn parse_libs_cflags(&mut self, name: &str, output: &[u8], config: &Config) {
+        #[cfg(all(windows, target_env = "msvc"))]
+        lazy_static::lazy_static! {
+            static ref LIB_BASENAME: regex::Regex = regex::Regex::new(r"^(.*)\.(lib|dll)$").unwrap();
+        }
+        #[cfg(all(windows, target_env = "gnu"))]
+        lazy_static::lazy_static! {
+            static ref LIB_BASENAME: regex::Regex = regex::Regex::new(r"^lib(.*)\.(a|dll)$").unwrap();
```

Comment:
This is not entirely correct still:
  - For MinGW and MSVC you can always link to `foo.lib` and `libfoo.a`
  - For MinGW you can in addition link to `libfoo.dll.a` and `foo.dll` / `libfoo.dll`

CC @nirbheek who explained this to me. I hope I summarized this correctly.

Reply:
For the `libfoo.a`, `libfoo.dll.a`, and `libfoo.dll` cases, does rustc-link-lib require `libfoo` or `lib`?

Reply:
You mean `libfoo` or `foo`? I'd guess it doesn't care either way